### ISSUE TITLE
OpenVPN Wintun driver is not compatitible with standard Wintun driver form Wireguard used in AdguardVPN

### ISSUE THREAD

**Issue by shamarin** [OpenVPN/openvpn-build#720]
**IMPORTANT NOTE**
Bugs about OpenVPN Access Server, OpenVPN Connect or any other product by OpenVPN Inc. should be directly reported to OpenVPN Inc. at https://support.openvpn.net

**Describe the bug**
OpenVPN Wintun driver is not compatitible with standard Wintun driver form Wireguard used in AdguardVPN (wintun.sys). You are using patched library form Wireguard and thats why now I have conflict with AdguardVPN. If i enable using of Wintun in AdguardVPN all my Wintun adapters get deleted by Adgurd.
https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661
Adguard advise you to use another name for Wintun adapter in Windows wintun.dll library that you are using.

**Version information (please complete the following information):**
 - OS: Windows 11 x64 22635.1076
 - OpenVPN version: 2.6.11
 
**Additional context**
Watch the last comment from Adguard developers https://github.com/AdguardTeam/AdGuardVPNForWindows/issues/661

**Comment by lstipakov** [comment]
JFTR: OpenVPN Wintun driver is a Wintun version 0.8.1, and indeed by no means patched.

**Comment by shamarin** [comment]
> JFTR: OpenVPN Wintun driver is a Wintun version 0.8.1, and indeed by no means patched.

Yes, OpenVPN using Wintun driver version 0.8.1 and Adguard VPN using newer version 1.14.0. As I understand AdgurdVPN devs the problem is that both are with the same name and using the same lib **wintun.sys** in the same path of Windows.
Will it be fixed?

**Comment by cron2** [comment]
> And how to be??? Now i can't use OpenVPN with Wintun and at the same time AdguardVPN with it's Wintun. If i connect to VPN server with AdguardVPN then OpenVPN tunnel being removed from Windows device manager. Even if i install it again then each time i connecting to AdguardVPN it's removing.

As @schwabe wrote - OpenVPN recommends to use dco-win instead of wintun since 2.6.0. 

Since the wintun issues are stemming from restrictions imposed on us by the wintun developers, it's unlikely that we're going to invest developer time into working around that problem (or that we could even fix it if we had the time).

**Comment by shamarin** [comment]
> > And how to be??? Now i can't use OpenVPN with Wintun and at the same time AdguardVPN with it's Wintun. If i connect to VPN server with AdguardVPN then OpenVPN tunnel being removed from Windows device manager. Even if i install it again then each time i connecting to AdguardVPN it's removing.
> 
> As @schwabe wrote - OpenVPN recommends to use dco-win instead of wintun since 2.6.0.
> 
> Since the wintun issues are stemming from restrictions imposed on us by the wintun developers, it's unlikely that we're going to invest developer time into working around that problem (or that we could even fix it if we had the time).

What are the benefits (speed) of dco-win compared to Wintun???

**Comment by shamarin** [comment]
> > > And how to be??? Now i can't use OpenVPN with Wintun and at the same time AdguardVPN with it's Wintun. If i connect to VPN server with AdguardVPN then OpenVPN tunnel being removed from Windows device manager. Even if i install it again then each time i connecting to AdguardVPN it's removing.
> > 
> > 
> > As @schwabe wrote - OpenVPN recommends to use dco-win instead of wintun since 2.6.0.
> > Since the wintun issues are stemming from restrictions imposed on us by the wintun developers, it's unlikely that we're going to invest developer time into working around that problem (or that we could even fix it if we had the time).
> 
> What are the benefits (speed) of dco-win compared to Wintun???

Founded myself. Will try new ovpn-dco-win driver.
